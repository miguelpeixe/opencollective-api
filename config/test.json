{
  "database": {
    "url": "postgres://opencollective@127.0.0.1:5432/opencollective_test"
  },
  "log": {
    "level": "warn"
  }
}
