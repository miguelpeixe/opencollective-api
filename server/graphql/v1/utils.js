import debug from 'debug';
import { graphql } from 'graphql';
import { loaders } from '../loaders';
import schema from './schema';

export const makeRequest = (remoteUser, query) => {
  return {
    remoteUser,
    body: { query },
    loaders: loaders({ remoteUser }),
  };
};

export const graphqlQuery = async (query, variables, remoteUser) => {
  const prepare = () => {
    if (remoteUser) {
      remoteUser.rolesByCollectiveId = null; // force refetching the roles
      return remoteUser.populateRoles();
    } else {
      return Promise.resolve();
    }
  };

  if (process.env.DEBUG && process.env.DEBUG.match(/graphql/)) {
    debug('graphql')('query', query);
    debug('graphql')('variables', variables);
    debug('graphql')('context', remoteUser);
  }

  return prepare().then(() =>
    graphql(
      schema,
      query,
      null, // rootValue
      makeRequest(remoteUser, query), // context
      variables,
    ),
  );
};
