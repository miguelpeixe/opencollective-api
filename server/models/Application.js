import crypto from 'crypto';
import { merge } from 'lodash';

export default function(Sequelize, DataTypes) {
  const Application = Sequelize.define(
    'Application',
    {
      id: {
        type: DataTypes.INTEGER,
        primaryKey: true,
        autoIncrement: true,
      },
      CollectiveId: {
        type: DataTypes.INTEGER,
        references: {
          model: 'Collectives',
          key: 'id',
        },
        onDelete: 'SET NULL',
        onUpdate: 'CASCADE',
      },
      CreatedByUserId: {
        type: DataTypes.INTEGER,
        references: {
          model: 'Users',
          key: 'id',
        },
        onDelete: 'SET NULL',
        onUpdate: 'CASCADE',
      },
      type: {
        type: DataTypes.ENUM,
        values: ['apiKey', 'oAuth'],
      },
      apiKey: {
        type: DataTypes.STRING,
      },
      clientId: {
        type: DataTypes.STRING,
      },
      clientSecret: {
        type: DataTypes.STRING,
      },
      callbackUrl: {
        type: DataTypes.STRING,
      },
      name: {
        type: DataTypes.STRING,
      },
      description: {
        type: DataTypes.STRING,
      },
      disabled: {
        type: DataTypes.BOOLEAN,
        defaultValue: false,
      },
      createdAt: {
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
      },
      updatedAt: {
        type: DataTypes.DATE,
        defaultValue: DataTypes.NOW,
      },
      deletedAt: {
        type: DataTypes.DATE,
      },
    },
    {
      paranoid: true,

      getterMethods: {
        info() {
          return {
            name: this.name,
            description: this.description,
            apiKey: this.apiKey,
            clientId: this.clientId,
            clientSecret: this.clientSecret,
            callbackUrl: this.callbackUrl,
          };
        },
      },
    },
  );

  Application.create = props => {
    if (props.type === 'apiKey') {
      props = merge(props, {
        apiKey: crypto.randomBytes(20).toString('hex'),
      });
    }
    if (props.type === 'oAuth') {
      props = merge(props, {
        clientId: crypto.randomBytes(20).toString('hex'),
        clientSecret: crypto.randomBytes(40).toString('hex'),
      });
    }
    return Application.build(props).save();
  };

  return Application;
}
