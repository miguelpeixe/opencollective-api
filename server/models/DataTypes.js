export default function(DataTypes) {
  return {
    // 3 letter international code (in uppercase) of the currency (e.g. USD, EUR, MXN, GBP, ...)
    currency: {
      type: DataTypes.STRING(3),
      defaultValue: 'USD',
      validate: {
        len: [3, 3],
      },
      allowNull: false,
      set(val) {
        if (val && val.toUpperCase) {
          this.setDataValue('currency', val.toUpperCase());
        }
      },
    },
  };
}
