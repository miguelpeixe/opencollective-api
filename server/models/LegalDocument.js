export default function(Sequelize, DataTypes) {
  const NOT_REQUESTED = 'NOT_REQUESTED';
  const REQUESTED = 'REQUESTED';
  const RECEIVED = 'RECEIVED';
  const ERROR = 'ERROR';

  const US_TAX_FORM = 'US_TAX_FORM';

  const LegalDocument = Sequelize.define(
    'LegalDocument',
    {
      id: {
        type: DataTypes.INTEGER,
        primaryKey: true,
        autoIncrement: true,
      },
      year: {
        type: DataTypes.INTEGER,
        validate: {
          min: 2015,
          notNull: true,
        },
        allowNull: false,
        unique: 'yearTypeCollective',
      },
      documentType: {
        type: DataTypes.ENUM,
        values: [US_TAX_FORM],
        allowNull: false,
        defaultValue: US_TAX_FORM,
        unique: 'yearTypeCollective',
      },
      documentLink: {
        type: DataTypes.STRING,
      },
      requestStatus: {
        type: DataTypes.ENUM,
        values: [NOT_REQUESTED, REQUESTED, RECEIVED, ERROR],
        allowNull: false,
        defaultValue: NOT_REQUESTED,
      },
      createdAt: {
        type: DataTypes.DATE,
        defaultValue: Sequelize.NOW,
      },
      updatedAt: {
        type: DataTypes.DATE,
        defaultValue: Sequelize.NOW,
      },
      deletedAt: {
        type: DataTypes.DATE,
      },
      CollectiveId: {
        type: DataTypes.INTEGER,
        references: {
          model: 'Collectives',
          key: 'id',
        },
        onDelete: 'SET NULL',
        onUpdate: 'CASCADE',
        allowNull: false,
        unique: 'yearTypeCollective',
      },
    },
    {
      paranoid: true,
    },
  );

  LegalDocument.findByTypeYearUser = ({ documentType, year, user }) => {
    return user.getCollective().then(collective => {
      return LegalDocument.findOne({
        where: {
          year,
          CollectiveId: collective.id,
          documentType,
        },
      });
    });
  };

  LegalDocument.hasUserCompletedDocument = async ({ documentType, year, user }) => {
    const doc = await LegalDocument.findByTypeYearUser({ documentType, year, user });

    return doc !== null && doc.requestStatus == RECEIVED;
  };

  LegalDocument.doesUserNeedToBeSentDocument = async ({ documentType, year, user }) => {
    const doc = await LegalDocument.findByTypeYearUser({ documentType, year, user });

    return doc == null || doc.requestStatus == NOT_REQUESTED || doc.requestStatus == ERROR;
  };

  LegalDocument.requestStatus = {
    REQUESTED,
    NOT_REQUESTED,
    RECEIVED,
    ERROR,
  };

  LegalDocument.associate = m => {
    LegalDocument.belongsTo(m.Collective, {
      foreignKey: 'CollectiveId',
      as: 'collective',
    });
  };

  return LegalDocument;
}
