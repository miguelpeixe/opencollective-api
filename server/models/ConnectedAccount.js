import config from 'config';

/**
 * Model.
 */
export default (Sequelize, DataTypes) => {
  const supportedServices = ['paypal', 'stripe', 'github', 'twitter', 'meetup'];

  const ConnectedAccount = Sequelize.define(
    'ConnectedAccount',
    {
      service: {
        type: DataTypes.STRING,
        validate: {
          isIn: {
            args: [supportedServices],
            msg: `Must be in ${supportedServices}`,
          },
        },
      },

      username: DataTypes.STRING, // paypal email / Stripe UserId / Twitter username / ...

      clientId: DataTypes.STRING, // paypal app id

      // either paypal secret OR an accessToken to do requests to the provider on behalf of the user
      token: DataTypes.STRING,
      refreshToken: DataTypes.STRING, // used for Stripe

      data: DataTypes.JSON, // Extra service provider specific data, e.g. Stripe: { publishableKey, scope, tokenType }
      settings: DataTypes.JSON, // configuration settings, e.g. defining templates for auto-tweeting

      createdAt: {
        type: DataTypes.DATE,
        defaultValue: Sequelize.NOW,
      },

      updatedAt: {
        type: DataTypes.DATE,
        defaultValue: Sequelize.NOW,
      },
    },
    {
      paranoid: true,

      getterMethods: {
        info() {
          return {
            id: this.id,
            service: this.service,
            username: this.username,
            createdAt: this.createdAt,
            updatedAt: this.updatedAt,
          };
        },

        paypalConfig() {
          return {
            client_id: this.clientId,
            client_secret: this.token,
            mode: config.paypal.rest.mode,
          };
        },
      },
    },
  );

  ConnectedAccount.associate = m => {
    ConnectedAccount.belongsTo(m.Collective, {
      foreignKey: 'CollectiveId',
      as: 'collective',
    });
  };

  return ConnectedAccount;
};
