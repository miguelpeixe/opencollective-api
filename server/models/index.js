import pg from 'pg';
import Sequelize from 'sequelize';
import config from 'config';
import debug from 'debug';

import logger from '../lib/logger';
import { getDBConf } from '../lib/db';

// this is needed to prevent sequelize from converting integers to strings, when model definition isn't clear
// like in case of the key totalOrders and raw query (like User.getTopBackers())
pg.defaults.parseInt8 = true;

const dbConfig = getDBConf('database');

/**
 * Database connection.
 */
logger.info(`Connecting to postgres://${dbConfig.host}/${dbConfig.database}`);

// If we launch the process with DEBUG=psql, we log the postgres queries
if (process.env.DEBUG && process.env.DEBUG.match(/psql/)) {
  config.database.options.logging = true;
}

if (config.database.options.logging) {
  if (process.env.NODE_ENV === 'production') {
    config.database.options.logging = (query, executionTime) => {
      if (executionTime > 50) {
        debug('psql')(query.replace(/(\n|\t| +)/g, ' ').slice(0, 100), '|', executionTime, 'ms');
      }
    };
  } else {
    config.database.options.logging = (query, executionTime) => {
      debug('psql')(
        '\n-------------------- <query> --------------------\n',
        query,
        `\n-------------------- </query executionTime="${executionTime}"> --------------------\n`,
      );
    };
  }
}

if (config.database.options.pool) {
  if (config.database.options.pool.min) {
    config.database.options.pool.min = parseInt(config.database.options.pool.min, 10);
  }
  if (config.database.options.pool.max) {
    config.database.options.pool.max = parseInt(config.database.options.pool.max, 10);
  }
}

export const sequelize = new Sequelize(dbConfig.database, dbConfig.username, dbConfig.password, {
  host: dbConfig.host,
  port: dbConfig.port,
  dialect: dbConfig.dialect,
  ...config.database.options,
});

const models = setupModels(sequelize);

export default models;

/**
 * Separate function to be able to use in scripts
 */
export function setupModels(client) {
  const m = {}; // models

  /**
   * Models.
   */

  [
    'Activity',
    'Application',
    'ConnectedAccount',
    'Collective',
    'Comment',
    'Expense',
    'LegalDocument',
    'Member',
    'Notification',
    'Order',
    'PaymentMethod',
    'RequiredLegalDocument',
    'Session',
    'Subscription',
    'Tier',
    'Transaction',
    'Update',
    'User',
  ].forEach(model => {
    m[model] = client.import(`${__dirname}/${model}`);
  });

  /**
   * Relationships
   */

  // PaymentMethod.
  m.PaymentMethod.belongsTo(m.Collective);
  m.PaymentMethod.belongsTo(m.PaymentMethod, {
    as: 'sourcePaymentMethod',
    foreignKey: 'SourcePaymentMethodId',
  });

  // User
  m.User.hasMany(m.Activity);
  m.User.hasMany(m.Notification);
  m.User.hasMany(m.Transaction, {
    foreignKey: 'CreatedByUserId',
    as: 'transactions',
  });
  m.User.hasMany(m.Order, { foreignKey: 'CreatedByUserId', as: 'orders' });
  m.User.hasMany(m.PaymentMethod, { foreignKey: 'CreatedByUserId' });
  m.User.hasMany(m.Member, { foreignKey: 'CreatedByUserId' });
  m.User.hasMany(m.ConnectedAccount, { foreignKey: 'CreatedByUserId' });
  m.User.belongsTo(m.Collective, {
    as: 'collective',
    foreignKey: 'CollectiveId',
    constraints: false,
  });

  // Members
  m.Member.belongsTo(m.User, {
    foreignKey: 'CreatedByUserId',
    as: 'createdByUser',
  });
  m.Member.belongsTo(m.Collective, {
    foreignKey: 'MemberCollectiveId',
    as: 'memberCollective',
  });
  m.Member.belongsTo(m.Collective, {
    foreignKey: 'CollectiveId',
    as: 'collective',
  });
  m.Member.belongsTo(m.Tier);

  // Activity.
  m.Activity.belongsTo(m.Collective);
  m.Activity.belongsTo(m.User);
  m.Activity.belongsTo(m.Transaction);

  // Notification.
  m.Notification.belongsTo(m.User);

  m.Notification.belongsTo(m.Collective);

  // Transaction.
  m.Collective.hasMany(m.Transaction, { foreignKey: 'CollectiveId' });
  m.Transaction.belongsTo(m.Collective, {
    foreignKey: 'CollectiveId',
    as: 'collective',
  });
  m.Transaction.belongsTo(m.Collective, {
    foreignKey: 'FromCollectiveId',
    as: 'fromCollective',
  });
  m.Transaction.belongsTo(m.Collective, {
    foreignKey: 'UsingVirtualCardFromCollectiveId',
    as: 'usingVirtualCardFromCollective',
  });

  m.Transaction.belongsTo(m.User, {
    foreignKey: 'CreatedByUserId',
    as: 'createdByUser',
  });
  m.Transaction.belongsTo(m.Collective, {
    foreignKey: 'HostCollectiveId',
    as: 'host',
  });
  m.Transaction.belongsTo(m.PaymentMethod);
  m.PaymentMethod.hasMany(m.Transaction);

  // Expense
  m.Expense.belongsTo(m.User);
  m.Expense.belongsTo(m.Collective, {
    foreignKey: 'CollectiveId',
    as: 'collective',
  });
  m.Transaction.belongsTo(m.Expense);
  m.Transaction.belongsTo(m.Order);

  // Order.
  m.Order.belongsTo(m.User, {
    foreignKey: 'CreatedByUserId',
    as: 'createdByUser',
  });
  m.Order.belongsTo(m.Collective, {
    foreignKey: 'FromCollectiveId',
    as: 'fromCollective',
  });
  m.Order.belongsTo(m.Collective, {
    foreignKey: 'CollectiveId',
    as: 'collective',
  });
  m.Order.belongsTo(m.Collective, {
    foreignKey: 'ReferralCollectiveId',
    as: 'referral',
  });
  m.Order.belongsTo(m.Tier);
  // m.Collective.hasMany(m.Order); // makes the test `mocha test/graphql.transaction.test.js -g "insensitive" fail
  m.Collective.hasMany(m.Member, { foreignKey: 'CollectiveId', as: 'members' });
  m.Collective.hasMany(m.Order, { foreignKey: 'CollectiveId', as: 'orders' });
  m.Transaction.belongsTo(m.Order);
  m.Order.hasMany(m.Transaction);
  m.Tier.hasMany(m.Order);

  // Subscription
  m.Order.belongsTo(m.Subscription); // adds SubscriptionId to the Orders table
  m.Subscription.hasOne(m.Order);

  // PaymentMethod
  m.Order.belongsTo(m.PaymentMethod, {
    foreignKey: 'PaymentMethodId',
    as: 'paymentMethod',
  });
  m.PaymentMethod.hasMany(m.Order);
  m.Transaction.belongsTo(m.PaymentMethod);

  // Tier
  m.Tier.belongsTo(m.Collective);

  Object.keys(m).forEach(modelName => m[modelName].associate && m[modelName].associate(m));

  return m;
}

export const Op = Sequelize.Op;
