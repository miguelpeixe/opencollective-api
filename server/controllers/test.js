import models from '../models';
import errors from '../lib/errors';
import { exportToPDF } from '../lib/utils';
import { getTransactions } from '../lib/transactions';
import moment from 'moment';

const envsAndDatabases = {
  development: 'opencollective_test',
  circleci: 'circle_test',
};

export const getTestUserLoginUrl = function(req, res, next) {
  if (envsAndDatabases[process.env.NODE_ENV]) {
    return models.User.findOne({
      where: { email: 'testuser@opencollective.com' },
    })
      .then(user => user.generateLoginLink(''))
      .then(link => link.replace('signin', 'login'))
      .then(link => res.redirect(link))
      .catch(next);
  }
  return next(
    new errors.BadRequest(`Unsupported NODE_ENV ${process.env.NODE_ENV} for retreiving test API login token`),
  );
};

export const exportPDF = function(req, res, next) {
  if (!envsAndDatabases[process.env.NODE_ENV]) {
    return next(new errors.BadRequest(`Unsupported NODE_ENV ${process.env.NODE_ENV} for testing export PDF`));
  }

  const d = new Date();
  d.setMonth(d.getMonth() - 1);
  const month = moment(d).format('MMMM');

  const startDate = new Date(d.getFullYear(), d.getMonth(), 1);
  const endDate = new Date(d.getFullYear(), d.getMonth() + 1, 1);

  console.log('startDate', startDate, 'endDate', endDate);

  const paper = req.query.papaer || 'Letter';
  const format = req.query.format || 'html';
  const wwcodeids = [
    '524',
    '47',
    '292',
    '275',
    '521',
    '525',
    '522',
    '262',
    '51',
    '295',
    '280',
    '283',
    '286',
    '510',
    '14',
    '515',
    '516',
    '518',
    '519',
    '520',
    '523',
    '512',
    '511',
    '513',
    '517',
    '59',
    '584',
    '299',
    '430',
    '48',
    '260',
    '261',
    '298',
    '272',
    '293',
    '273',
    '294',
    '263',
    '274',
    '276',
    '277',
    '301',
    '195',
    '241',
    '265',
    '297',
    '259',
    '266',
    '279',
    '267',
    '278',
    '12',
    '269',
    '270',
    '281',
    '10',
    '282',
    '3',
    '284',
    '264',
    '287',
    '268',
    '4',
    '300',
    '289',
    '13',
    '291',
    '285',
    '288',
    '271',
    '290',
    '15',
    '2',
  ];
  getTransactions(wwcodeids, startDate, endDate, {
    where: { type: 'DEBIT' },
    include: ['User', 'Expense', 'Collective'],
  }).then(transactions => {
    console.log('transactions', JSON.stringify(transactions));
    const data = {
      host: { name: 'WWCode', currency: 'USD' },
      year: new Date().getFullYear(),
      month,
    };
    let page = 1;
    let currentPage = 0;
    const transactionsPerTOCPage = 30; // number of transaction per page of the Table Of Content (for PDF export)
    const note = 'using fxrate of the day of the transaction as provided by the ECB. Your effective fxrate may vary.';
    data.expensesPerPage = [[]];
    data.totalPaidExpenses = transactions.length;
    data.transactions = transactions.map(t => {
      t.page = page++;
      t.collective = t.Collective;
      t.collective.shortSlug = t.collective.slug.replace(/^wwcode-?(.)/, '$1');
      t.privateMessage = t.Expense && t.Expense.privateMessage;
      if (t.data && t.data.fxrateSource) {
        t.privateMessage = t.privateMessage ? `${t.privateMessage} (${note})` : note;
        data.privateMessage = note;
      }
      if (page - (1 % transactionsPerTOCPage) === 0) {
        currentPage++;
        data.expensesPerPage[currentPage] = [];
      }
      data.expensesPerPage[currentPage].push(t);
      return t;
    });
    exportToPDF('expenses', data, { format, paper }).then(html => {
      if (format === 'pdf') {
        res.setHeader('content-type', 'application/pdf');
      }
      res.send(html);
    });
  });
};
