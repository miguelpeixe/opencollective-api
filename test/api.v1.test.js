/**
 * Note: to update the snapshots run:
 * TZ=UTC CHAI_JEST_SNAPSHOT_UPDATE_ALL=true npx mocha test/api.v1.test.js
 */

import request from 'supertest';
import { expect } from 'chai';
import { omit, orderBy } from 'lodash';

import app from '../server/index';
import models from '../server/models';
import roles from '../server/constants/roles';
import * as utils from '../test/utils';

const application = utils.data('application');
const publicCollectiveData = utils.data('collective1');
const transactionsData = utils.data('transactions1.transactions');

const orderTransactions = transactions => orderBy(transactions, ['createdAt', 'id'], ['desc', 'desc']);

describe('api.v1.test.js', () => {
  let collective, user, host;

  before(() => utils.resetTestDB());

  // Create users
  before('create user', () => models.User.createUserWithCollective(utils.data('user1')).tap(u => (user = u)));
  before('create host', () => models.User.createUserWithCollective(utils.data('host1')).tap(u => (host = u)));

  // Create collectives
  before('create collective', () => models.Collective.create(publicCollectiveData).tap(g => (collective = g)));

  // Add users to collectives
  before('add host to collective', () => collective.addHost(host.collective));
  before('add user to collective as a member', () => collective.addUserWithRole(user, roles.ADMIN));

  let transaction, defaultAttributes;

  before(() => {
    defaultAttributes = {
      CreatedByUserId: user.id,
      FromCollectiveId: user.CollectiveId,
      HostCollectiveId: host.CollectiveId,
    };
  });

  before('create multiple transactions for collective', () => {
    return models.Transaction.createMany(transactionsData, {
      CollectiveId: collective.id,
      ...defaultAttributes,
    })
      .then(orderTransactions)
      .then(transactions => {
        transaction = transactions[0];
      });
  });

  /**
   * Get collective's transactions.
   */
  describe('#get', () => {
    it('get the latest transactions', done => {
      request(app)
        .get(`/v1/collectives/${collective.slug}/transactions?api_key=${application.api_key}`)
        .expect(200)
        .end((e, res) => {
          expect(e).to.not.exist;
          const transactions = orderTransactions(res.body.result);
          expect(transactions.length).to.equal(9);
          const debits = transactions.filter(t => t.type === 'DEBIT');
          const credits = transactions.filter(t => t.type === 'CREDIT');
          expect(omit(debits[0], ['id', 'createdAt'])).to.matchSnapshot();
          expect(omit(credits[0], ['id', 'createdAt'])).to.matchSnapshot();
          done();
        });
    });

    it('get one transaction details by uuid', done => {
      request(app)
        .get(`/v1/collectives/${collective.slug}/transactions/${transaction.uuid}?api_key=${application.api_key}`)
        .expect(200)
        .end((e, res) => {
          expect(e).to.not.exist;
          const transaction = omit(res.body.result, ['id', 'createdAt']);
          expect(transaction).to.matchSnapshot();
          done();
        });
    });
  });
});
