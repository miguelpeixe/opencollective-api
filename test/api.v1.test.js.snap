// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api.test.js #get get one transaction details by uuid 1`] = `
Object {
  "amount": -20000,
  "collective": Object {
    "id": 3,
    "image": null,
    "name": "Scouts d'Arlon",
    "slug": "scouts",
  },
  "currency": "USD",
  "description": "Homepage design",
  "fromCollective": Object {
    "id": 1,
    "image": null,
    "name": "Phil Mod",
    "slug": "philmod",
  },
  "host": Object {
    "id": 2,
    "image": null,
    "name": "WWCode",
    "slug": "wwcode",
  },
  "hostCurrency": "USD",
  "hostCurrencyFxRate": null,
  "hostFeeInHostCurrency": null,
  "netAmountInCollectiveCurrency": null,
  "paymentMethod": null,
  "paymentProcessorFeeInHostCurrency": null,
  "platformFeeInHostCurrency": null,
  "type": "DEBIT",
  "uuid": null,
}
`;

exports[`api.test.js #get get the latest transactions 1`] = `
Object {
  "amount": 5000,
  "collective": Object {
    "id": 3,
    "image": null,
    "name": "Scouts d'Arlon",
    "slug": "scouts",
  },
  "createdByUser": Object {
    "email": null,
    "id": 1,
  },
  "currency": "USD",
  "fromCollective": Object {
    "id": 1,
    "image": null,
    "name": "Phil Mod",
    "slug": "philmod",
  },
  "host": Object {
    "id": 2,
    "slug": "wwcode",
  },
  "hostCurrency": null,
  "hostCurrencyFxRate": null,
  "hostFeeInHostCurrency": null,
  "netAmountInCollectiveCurrency": 4500,
  "paymentMethod": null,
  "paymentProcessorFeeInHostCurrency": null,
  "platformFeeInHostCurrency": null,
  "type": "CREDIT",
  "uuid": null,
}
`;

exports[`api.v1.test.js #get get one transaction details by uuid 1`] = `
Object {
  "amount": 5000,
  "collective": Object {
    "id": 3,
    "image": null,
    "name": "Scouts d'Arlon",
    "slug": "scouts",
  },
  "currency": "USD",
  "description": "Donation to that great project",
  "fromCollective": Object {
    "id": 1,
    "image": null,
    "name": "Phil Mod",
    "slug": "philmod",
  },
  "host": Object {
    "id": 2,
    "image": null,
    "name": "WWCode",
    "slug": "wwcode",
  },
  "hostCurrency": null,
  "hostCurrencyFxRate": null,
  "hostFeeInHostCurrency": null,
  "netAmountInCollectiveCurrency": 4500,
  "order": null,
  "paymentMethod": null,
  "paymentProcessorFeeInHostCurrency": null,
  "platformFeeInHostCurrency": null,
  "type": "CREDIT",
  "uuid": null,
}
`;

exports[`api.v1.test.js #get get the latest transactions 1`] = `
Object {
  "amount": -1199,
  "collective": Object {
    "id": 3,
    "image": null,
    "name": "Scouts d'Arlon",
    "slug": "scouts",
  },
  "createdByUser": Object {
    "email": null,
    "id": 1,
  },
  "currency": "USD",
  "fromCollective": Object {
    "id": 1,
    "image": null,
    "name": "Phil Mod",
    "slug": "philmod",
  },
  "host": Object {
    "id": 2,
    "slug": "wwcode",
  },
  "hostCurrency": null,
  "hostCurrencyFxRate": null,
  "hostFeeInHostCurrency": null,
  "netAmountInCollectiveCurrency": null,
  "paymentMethod": null,
  "paymentProcessorFeeInHostCurrency": null,
  "platformFeeInHostCurrency": null,
  "type": "DEBIT",
  "uuid": null,
}
`;

exports[`api.v1.test.js #get get the latest transactions 2`] = `
Object {
  "amount": 5000,
  "collective": Object {
    "id": 3,
    "image": null,
    "name": "Scouts d'Arlon",
    "slug": "scouts",
  },
  "createdByUser": Object {
    "email": null,
    "id": 1,
  },
  "currency": "USD",
  "fromCollective": Object {
    "id": 1,
    "image": null,
    "name": "Phil Mod",
    "slug": "philmod",
  },
  "host": Object {
    "id": 2,
    "slug": "wwcode",
  },
  "hostCurrency": null,
  "hostCurrencyFxRate": null,
  "hostFeeInHostCurrency": null,
  "netAmountInCollectiveCurrency": 4500,
  "paymentMethod": null,
  "paymentProcessorFeeInHostCurrency": null,
  "platformFeeInHostCurrency": null,
  "type": "CREDIT",
  "uuid": null,
}
`;
